src/System.Collections.Specialized/src/System/Web/HttpUtility.cs

```diff
@@ -0,0 +1,348 @@
+                query = query.Substring(1);
+
+            var result = new HttpQSCollection();
+            ParseQueryString(query, encoding, result);
+            return result;
+        }
+
+        private static void ParseQueryString(string query, Encoding encoding, NameValueCollection result)
+        {
+            if (query.Length == 0)
+                return;
+
+            var decoded = HtmlDecode(query);
+            var decodedLength = decoded.Length;
+            var namePos = 0;
+            var first = true;
+            while (namePos <= decodedLength)
+            {
+                int valuePos = -1, valueEnd = -1;
+                for (var q = namePos; q < decodedLength; q++)
+                    if ((valuePos == -1) && (decoded[q] == '='))
+                    {
+                        valuePos = q + 1;
+                    }
+                    else if (decoded[q] == '&')
+                    {
+                        valueEnd = q;
+                        break;
+                    }
+
+                if (first)
+                {
+                    first = false;
+                    if (decoded[namePos] == '?')
+                        namePos++;
+                }
+
+                string name;
+                if (valuePos == -1)
+                {
+                    name = null;
+                    valuePos = namePos;
+                }
+                else
+                {
+                    name = UrlDecode(decoded.Substring(namePos, valuePos - namePos - 1), encoding);
+                }
+                if (valueEnd < 0)
+                {
+                    namePos = -1;
+                    valueEnd = decoded.Length;
+                }
+                else
+                {
+                    namePos = valueEnd + 1;
+                }
+                var value = UrlDecode(decoded.Substring(valuePos, valueEnd - valuePos), encoding);
+
+                result.Add(name, value);
+                if (namePos == -1)
+                    break;
+            }
+        }
+
+        /// <devdoc>
+        ///     <para>
+        ///         HTML decodes a string and returns the decoded string.
+        ///     </para>
+        /// </devdoc>
+        public static string HtmlDecode(string s)
+        {
+            return HttpEncoder.HtmlDecode(s);
+        }
+
+
+        /// <devdoc>
+        ///     <para>
+        ///         HTML decode a string and send the result to a TextWriter output stream.
+        ///     </para>
+        /// </devdoc>
+        public static void HtmlDecode(string s, TextWriter output)
+        {
+            HttpEncoder.HtmlDecode(s, output);
+        }
+
+
+        /// <devdoc>
+        ///     <para>
+        ///         HTML encodes a string and returns the encoded string.
+        ///     </para>
+        /// </devdoc>
+        public static string HtmlEncode(string s)
+        {
+            return HttpEncoder.HtmlEncode(s);
```

Comment:
Wouldn't this just call WebUtility.HtmlEncode? As @davidsh was pointing out? Similarly for the other overloads and UrlEncode.